src/components/map/map.tsx

```diff
@@ -0,0 +1,234 @@
+];
+
+export function Map({
+  position = defaultPosition,
+  layer,
+  onSelectStopPlace,
+}: MapProps) {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const map = useRef<mapboxgl.Map>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (!mapContainer.current) return;
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      accessToken: mapboxData.accessToken,
+      style: mapboxData.style,
+      center: position,
+      zoom: 13,
+    });
+
+    return () => map.current?.remove();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const { centerMap } = useMapInteractions(map, onSelectStopPlace);
+  const { mapWrapperRef, toggleFullscreen } = useFullscreenMap(map);
+  useMapPin(map, position, layer);
+
+  useEffect(() => {
+    if (map.current) {
+      map.current.flyTo({
+        center: position,
+        zoom: 15,
+        speed: 2,
+      });
+    }
+  }, [position]);
+
+  return (
+    <div className={style.map}>
+      <Button
+        className={style.fullscreenButton}
+        title={t(ComponentText.Map.map.openFullscreenButton)}
+        icon={{ left: <MonoIcon src="map/Map.svg" /> }}
+        onClick={toggleFullscreen}
+      />
+
+      <div className={style.mapWrapper} ref={mapWrapperRef}>
+        <Button
+          className={style.closeButton}
+          onClick={toggleFullscreen}
+          size="small"
+          icon={{
+            left: (
+              <MonoIcon src="navigation/ArrowLeft.svg" overrideColor="white" />
+            ),
+          }}
+          mode="interactive_0"
+        />
+        <Button
+          className={style.buttonsContainer}
+          size="small"
+          icon={{ left: <MonoIcon src="places/City.svg" /> }}
+          onClick={() => centerMap(position)}
+        />
+        <div ref={mapContainer} className={style.mapContainer} />
+      </div>
+    </div>
+  );
+}
+
+const defaultPosition: LngLatPosition = [
+  mapboxData.defaultLng,
+  mapboxData.defaultLat,
+];
+
+function useMapPin(
+  mapRef: React.MutableRefObject<mapboxgl.Map | undefined>,
+  position: LngLatPosition,
+  layer?: string,
+) {
+  const [marker, setMarker] = useState<mapboxgl.Marker | undefined>();
+
+  const createMapPin = useCallback(() => {
+    const pinSvg = React.createElement(
+      'span',
+      { className: style.positionMarker },
+      [<ColorIcon key="pin" src="Pin.svg" />],
+    ) as Marker;
+
+    return new mapboxgl.Marker(pinSvg);
+  }, []);
+
+  const setMapPinPosition = useCallback(
+    (position: LngLatPosition) => {
+      if (!mapRef || !mapRef.current) return;
+      const map = mapRef.current;
+      if (!marker) {
+        const newMarker = createMapPin();
+        setMarker(newMarker as mapboxgl.Marker);
+      }
+
+      if (marker) marker.setLngLat(position).addTo(map);
+    },
+    [createMapPin, mapRef, marker],
+  );
+
+  const removeMarkerFromMap = useCallback(() => {
+    if (!marker) return;
+    marker.remove();
```

Comment:
This can also be simplified to this

```suggestion
    marker?.remove();
```